### PR TITLE
applyconfig: support disabled clusters

### PR DESCRIPTION
**Pull request by hongkailiu**
Follow up https://issues.redhat.com/browse/DPTP-3659?focusedId=23220514&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-23220514

/cc @openshift/test-platform 
/assign @smg247

### CODE REVIEW

File: cmd/applyconfig/applyconfig.go [hunk 1]
```diff
@@ -571,6 +573,17 @@ func main() {
 	}
 	censor := secrets.NewDynamicCensor()
 	logrus.SetFormatter(logrusutil.NewFormatterWithCensor(logrus.StandardLogger().Formatter, &censor))
+
+	prowDisabledClusters, err := prowconfigutils.ProwDisabledClusters(nil)
+	if err != nil {
+		logrus.WithError(err).Warn("Failed to get Prow disable clusters")
+	} else {
+		if len(prowDisabledClusters) > 0 && targetClusterIsDisabled(sets.New[string](prowDisabledClusters...)) {
```
Comment (smg247) [inline cmd/applyconfig/applyconfig.go:581]:
```suggestion
	} else if len(prowDisabledClusters) > 0 && targetClusterIsDisabled(sets.New[string](prowDisabledClusters...)) {
```

File: cmd/applyconfig/applyconfig.go [hunk 2]
```diff
@@ -588,3 +601,17 @@ func main() {
 
 	logrus.Infof("Success!")
 }
+
+func targetClusterIsDisabled(disabledClusters sets.Set[string]) bool {
```
Comment (smg247) [inline cmd/applyconfig/applyconfig.go:605]: can we just do a `strings.Contains` check here to see if the jobName contains any of the `disabledClusters`? That way, if the job name is changed slightly we won't have to update this code.

Comment (hongkailiu) [reply cmd/applyconfig/applyconfig.go:603]: I did it like it the first time but changed my mind. :) 
Revised.

File: cmd/applyconfig/applyconfig.go (hunk 1 above)
Comment (hongkailiu) [reply cmd/applyconfig/applyconfig.go]: Done.

**Comment by hongkailiu** [comment]
Seems working.
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/44423/pull-ci-openshift-release-master-build09-dry/1714052061928099840

**Pull request merged**